src/Raven.Server/Documents/Indexes/MapReduce/Auto/AutoMapReduceIndexDefinition.cs

```diff
@@ -19,6 +19,16 @@ internal class AutoMapReduceIndexDefinition : AutoIndexDefinitionBaseServerSide
 
         public readonly AutoIndexField[] OrderedGroupByFields;
 
+        private List<string> _groupByFieldNames;
+
+        public List<string> GroupByFieldNames
+        {
+            get
+            {
+                return _groupByFieldNames ??= GroupByFields.Keys.ToList();
+            }
+        }
```

Comment:
I think that originally you had it as `HashSet`, now it's `List`. I assume you did it to preserve order. Is that correct? If that's the case then I don't think that `GroupByFields.Keys.ToList()` guarantees that. Note that we have `OrderedGroupByFields`

Anyway, I don't think that we should care about that on this level. If you need that somewhere then assure it there.

Reply:
Fixed. I'm looking for a consistent order (of any kind).
So if the group by keys in the function are something like `group by Name, LastName`, I want them in that order.

Reply:
`OrderedGroupByFields` doesn't preserve the original order.